python/lance/_lib.pyx

```diff
@@ -311,3 +306,116 @@ cdef class FileSystemDataset(Dataset):
 def _wrap_dataset(Dataset dataset not None):
     cdef shared_ptr[CDataset] copy = dataset.unwrap()
     return FileSystemDataset.wrap(move(copy))
+
+
+cdef extern from "lance/arrow/dataset.h" namespace "lance::arrow" nogil:
+    cdef cppclass CLanceDataset "::lance::arrow::LanceDataset":
+        enum WriteMode "WriteMode":
+            CREATE "::lance::arrow::LanceDataset::WriteMode::kCreate"
+            APPEND "::lance::arrow::LanceDataset::WriteMode::kAppend"
+            OVERWRITE "::lance::arrow::LanceDataset::WriteMode::kOverwrite"
+
+        @ staticmethod
+        CStatus Write(
+                const CFileSystemDatasetWriteOptions& write_options,
+                shared_ptr[CDataset] dataset,
+                WriteMode mode)
+
+        @ staticmethod
+        CResult[shared_ptr[CLanceDataset]] Make(
+                const shared_ptr[CFileSystem]& fs,
+                const string& base_uri,
+                optional[uint64_t] version,
+        )
+
+
+
+cdef class LanceDataset(Dataset):
+    """Lance Dataset.
+    """
+    cdef:
+        CLanceDataset * lance_dataset
+
+    def __init__(self):
+        _forbid_instantiation(self.__class__)
+
+    cdef void init(self, const shared_ptr[CDataset]& sp):
+        Dataset.init(self, sp)
+        self.lance_dataset = <CLanceDataset *> sp.get()
+
+    @staticmethod
+    cdef wrap(const shared_ptr[CDataset]& sp):
+        cdef LanceDataset ds = LanceDataset.__new__(LanceDataset)
+        ds.init(sp)
+        return ds
+
+    def head(self, n: int, offset: int = 0) -> Table:
```

Comment:
i think the superclass supports **kwargs. This makes it possible to pass in a `columns=` argument directly to `head` instead of having to do `ds.scanner(columns=[], limit=10).to_table()`. it's useful for our usecases so you can pick only a few non-blob columns

Reply:
fixed